tests/unit/retrievers/test_base.py

```diff
@@ -12,12 +12,14 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
+from __future__ import annotations  # Reminder: May be removed after Python 3.9 is EOL.
```

Comment:
Out of curiosity, why is this import required only in this file?

Reply:
We could do it for every file too tbh and we can drop it everywhere if we stop supporting Python 3.8 in future

Reply:
https://github.com/neo4j/neo4j-genai-python/pull/67/commits/46670c0fee52b942ff18a50c13b247b48e269595